src/main/java/org/testng/internal/Parameters.java

```diff
@@ -52,11 +52,39 @@
       AfterMethod.class
   };
 
-  private static Map<String, List<Class>> mapping = Maps.newHashMap();
+  private static Map<String, List<Class<?>>> mapping = Maps.newHashMap();
+/*
+          +--------------+--------------+---------+--------+----------+-------------+
```

Comment:
We should display a warning when `XmlTest` is used in a before suite.

But what about `XmlSuite`, `XmlClass`, ...? 
I think we can consider `ISuite`, `IClass`, ... too.